north/src/runtime/linux/device_mapper.rs

```diff
@@ -279,14 +291,19 @@ impl Dm {
     /// held until it is resumed.
     ///
     /// Valid flags: DM_SUSPEND, DM_NOFLUSH, DM_SKIP_LOCKFS
-    pub async fn device_suspend(&self, id: &str, options: &DmOptions) -> Result<DeviceInfo> {
+    pub async fn device_suspend(
+        &self,
+        id: &str,
+        options: &DmOptions,
+    ) -> Result<DeviceInfo, DeviceMapperError> {
         let mut hdr = options.to_ioctl_hdr(
             Some(id),
             DmFlags::DM_SUSPEND | DmFlags::DM_NOFLUSH | DmFlags::DM_SKIP_LOCKFS,
         );
 
         self.do_ioctl(DM_DEV_SUSPEND_CMD as u8, &mut hdr, None)
-            .await?;
+            .await
+            .map_err(|_| DeviceMapperError::SuspendDeviceError);
```

Comment:
This elides the original io::Error. `DeviceMapperError` should wrap a `io::Error`